lua/neotest-kotlin/init.lua

```diff
@@ -8,57 +8,54 @@ local filter = require("neotest-kotlin.src.filter")
 local treesitter_query = require("neotest-kotlin.kotest-treesitter-query")
 local output_parser = require("neotest-kotlin.src.output-parser")
 
-local adapter = { name = "neotest-kotest" }
+local M = {}
 
-function adapter.root(dir)
+---@class neotest.Adapter
+---@field name string
+M.Adapter = { name = "neotest-kotest" }
+
+---Find the project root directory given a current directory to work from.
+---Should no root be found, the adapter can still be used in a non-project context if a test file matches.
+---@async
+---@param dir string @Directory to treat as cwd
+---@return string | nil @Absolute root dir of test suite
+function M.Adapter.root(dir)
   return lib.files.match_root_pattern("gradlew")(dir)
 end
 
-function adapter.filter_dir(name, rel_path, root)
+---Filter directories when searching for test files
+---@async
+---@param name string Name of directory
+---@param rel_path string Path to directory, relative to root
+---@param root string Root directory of project
+---@return boolean
+function M.Adapter.filter_dir(name, rel_path, root)
   return filter.test_directory(name)
 end
 
-function adapter.is_test_file(file_path)
+---@async
+---@param file_path string
+---@return boolean
+function M.Adapter.is_test_file(file_path)
   return filter.is_test_file(file_path)
 end
 
-local function get_match_type(captured_nodes)
-  if captured_nodes["namespace.name"] then
-    return "namespace"
-  end
-  if captured_nodes["test.name"] then
-    return "test"
-  end
-end
-
-function adapter.build_position(file_path, source, captured_nodes)
-  local match_type = get_match_type(captured_nodes)
-  local definition = captured_nodes[match_type .. ".definition"]
-
-  local build_position = {
-    type = match_type,
-    path = file_path,
-    range = { definition:range() },
-  }
-
-  return build_position
-end
```

Comment:
This may be part of the old interface, but isn't part of it anymore and isn't used anywhere?